public/app/plugins/panel/canvas/editor/element/APIEditor.tsx

```diff
@@ -138,17 +188,33 @@ export function APIEditor({ value, context, onChange }: Props) {
           />
         </InlineField>
       </InlineFieldRow>
+      <InlineFieldRow>
+        <InlineField label="Method" labelWidth={LABEL_WIDTH} grow={true}>
+          <RadioButtonGroup value={value?.method} options={httpMethodOptions} onChange={onMethodChange} fullWidth />
+        </InlineField>
+      </InlineFieldRow>
+      <InlineFieldRow>
+        <InlineField label="Type" labelWidth={LABEL_WIDTH} tooltip="How the parameters are sent" grow={true}>
+          <RadioButtonGroup
+            value={value?.paramsType}
+            options={httpParamsType}
+            onChange={onParamsTypeChange}
+            fullWidth
+          />
+        </InlineField>
+      </InlineFieldRow>
+      <Field label="Parameters" description="Query parameters">
+        <QueryParamsEditor value={value?.params ?? []} onChange={onParamsChange} />
+      </Field>
       {value?.method === HttpRequestMethod.POST && (
-        <InlineFieldRow>
-          <InlineField label={'Data'} labelWidth={LABEL_WIDTH} grow={true}>
-            <StringValueEditor
-              context={context}
-              value={value?.data ?? '{}'}
-              onChange={onDataChange}
-              item={dummyStringSettings}
-            />
-          </InlineField>
-        </InlineFieldRow>
+        <Field label="Payload">
+          <StringValueEditor
+            context={context}
+            value={value?.data ?? '{}'}
+            onChange={onDataChange}
+            item={{ ...dummyStringSettings, settings: { useTextarea: true } }}
+          />
```

Comment:
It might be nice if we can figure out how to add the autocomplete functionality (i.e. when typing ${ the list of possible template variables come up) - I think I've seen this functionality somewhere else in our UI before but not sure exactly where 🤷